### DISCUSSION TITLE
Can I publish only parts of my Repository and keep other folders private?

### DISCUSSION THREAD

**Discussion by Santonia27** [community/community#81015] [Repositories] [answered]
### Select Topic Area

Question

### Body

Is it possible to publish the docs of my repo but keep the rest of the repository private? 
Are there difference among github subscriptions? 
thanks

**Comment by ananuness** [comment]
You can use the file `.gitignore` to ignore the files you don't want to expose. Example: 

You don't want to publish a `projects` folder that is inside of your repo, so you can add in the `.gitignore`: 

```
/projects

# or

projects/
```

Then, git will not show it as a file to commit and you don't need to make it public.

- [Ignoring files](https://git-scm.com/book/en/v2/Git-Basics-Recording-Changes-to-the-Repository#_ignoring)

- [Gitignore and .gitignore]( 
https://www.w3schools.com/git/git_ignore.asp?remote=github)

I hope it helps!

**Reply by Santonia27** [reply]
Hi Ana, thanks for the message. However, the repo should be visible within our organization so I need to commit and publish all the files within the organization. The repo is private within the organization but we would like to publish the docs to the public. 

So I believe .gitignore will not work in this case. 

Thanks anyway!

**Reply by ananuness** [reply]
Oh, I'm sorry I didn't understand it right 😅 In this case the only way you have to make it available is adding the docs to another repo! Even with the premium, there's not a feature that allows us to have some files private and some public in the same repo!! Sadly. It would be a great feature!

But you can use submodules to at least "connect" your private repo with the docs (the public repo). Or, of course, you can just link the docs in the readme of the private repo.

See more about submodules [here](https://gist.github.com/gitaarik/8735255).

**Comment by faridnec** [comment]
Here maybe you can try this?
- you can enable GitHub Pages for your private repository and place documents in the docs folder or custom location. Here is the link for more info [link](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site)
- The documentation will be accessible publicly at `https://<username>.github.io/<repo>`, even though the repository remains private.

**Reply by Santonia27** [reply]
okay thanks I will look into this! and write an update, if it worked!

**Reply by faridnec** [reply]
hey, can you share the update? and if it works you may mark the answer. Thanks :)

**Comment by duhoshina** [comment]
I'm having the same issue, would like to know if you managed to solve it, and if so, what did you do?

**Reply by ananuness** [reply, marked as answer]
As @faridnec said, you can create a page for the docs in GitHub pages in the private repo or as I said you can create a public repo for the docs and optionally create a submodule of it. The only con with GitHub pages is that you have to provide the link somewhere else, since the repo is private.

**Reply by duhoshina** [reply]
Shoow, Ana! Muito obrigado, vou fazer isso mesmo. :)

**Reply by ananuness** [reply]
Eita, de nada!! Espero que dê tudo certo!!
